ts/torch_handler/base_handler.py

```diff
@@ -73,6 +81,9 @@ def initialize(self, context):
             self.model = self._load_torchscript_model(model_pt_path)
 
         self.model.eval()
+	if ipex_enabled:
+            self.model = self.model.to(memory_format=torch.channels_last)
+            self.model = ipex.optimize(self.model, dtype=torch.float32, level='O1')
```

Comment:
> is there a document that describes what the ipex levels mean?

Yes, there will be a user guide as part of IPEX 1.10 release. But here, I would suggest we use the default "level" recommended by IPEX, not hard-code 'O1' here.

> Also what happens here if a user passes in a model that's already in float16 or bffloat16?

@min-jean-cho please remove the `dtype` arg. @msaroufim The default behavior without specifying `dtype` is do nothing. If the user specifies a low-precision data type (e.g., bfloat16 or float16, note that float16 is not supported now) different from the low-precision data type the model has, IPEX will do nothing and issue a warning message. The behavior will be clarified in the user doc too.

@min-jean-cho Suggest to change this line to `self.model = ipex.optimize(self.model)`

Reply:
@jgong5 it would also be great to clarify on different levels in the doc as well.

Reply:
@HamidShojanazeri Yes, this is in the plan.

Reply:
```self.model = ipex.optimize(self.model, dtype=torch.float32, level='O1')``` this also only works for fp32; So ```if ipex_enabled:``` is insufficient condition to check

Reply:
@min-jean-cho may I know the specific scenario you are concerned with if you change the code to `self.model = ipex.optimize(self.model)`?

Reply:
@jgong5 You're right. I'll integrate your two suggestions regarding ```dtype``` arg and ```self.model = ipex.optiize(self.model)``` - that would work.

Reply:
> @jgong5 You're right. I'll integrate your two suggestions regarding `dtype` arg and `self.model = ipex.optiize(self.model)` - that would work.

wouldn't we still need the level?

Reply:
We don't need to explicitly set the level as ```level = 'O1'``` since it is the default value and we are optimizing the model with ```ipex enabled```